integrations/oci/sdk/runtime/src/main/java/io/helidon/integrations/oci/sdk/runtime/OciExtension.java

```diff
@@ -109,44 +114,124 @@
                                     .map(OciAuthenticationDetailsProvider.AuthStrategy::id)
                                     .toList())
             .build());
+    private static String overrideOciConfigFile;
+    private static Supplier<io.helidon.common.config.Config> ociConfigSupplier;
 
     private OciExtension() {
     }
 
     /**
-     * Returns the {@link OciConfig} that is currently defined in the bootstrap environment. If one is not defined under
-     * config {@value OciConfig#CONFIG_KEY} then a default implementation will be constructed.
+     * The configured authentication provider strategy type name. Note, however, that the authentication strategy returned may not
+     * necessarily be available. The configured authentication provider merely returns what is configured via
+     * {@value OciAuthenticationDetailsProvider#KEY_AUTH_STRATEGY} and/or
+     * {@value OciAuthenticationDetailsProvider#KEY_AUTH_STRATEGIES}. In order to additionally check if the provider is available,
+     * the {@code verifyIsAvailable} argument should be {@code true}.
+     *
+     * @param verifyIsAvailable flag to indicate whether the provider should be checked for availability
+     * @return the configured authentication type name
+     */
+    public static Class<? extends AbstractAuthenticationDetailsProvider>
+    configuredAuthenticationDetailsProvider(boolean verifyIsAvailable) {
+        return OciAuthenticationDetailsProvider.select(ociConfig(), verifyIsAvailable).authStrategy().type();
+    }
+
+    /**
+     * Returns the global {@link OciConfig} bean that is currently defined in the bootstrap environment.
+     * <p>
+     * The implementation will first look for an {@code oci.yaml} file, and if found will use that file to establish the global
+     * oci-specific bootstrap {@link io.helidon.config.spi.ConfigSource}.
+     * <p>
+     * If the implementation is unable to find this file, then a fallback mechanism will be used to find it in the configuration
+     * found in the {@link InjectionServices#globalBootstrap()}, using a top-level attribute key named
+     * {@value OciConfigBlueprint#CONFIG_KEY}.
+     * <p>
+     * The final fallback mechanism will use an {@code auto} authentication strategy - see {@link OciConfigBlueprint} for details.
      *
      * @return the bootstrap oci config bean
+     * @see OciConfigBlueprint
+     * @see #ociConfigSupplier
      */
     public static OciConfig ociConfig() {
-        Optional<Bootstrap> bootstrap = InjectionServices.globalBootstrap();
-        if (bootstrap.isEmpty()) {
-            LOGGER.log(System.Logger.Level.DEBUG, "No bootstrap - using default oci config");
-            return DEFAULT_OCI_CONFIG_BEAN.get();
-        }
+        io.helidon.common.config.Config config = configSupplier().get();
+        if (!config.exists()
+                || !(config.get(OciAuthenticationDetailsProvider.KEY_AUTH_STRATEGY).exists()
```

Comment:
can you do static import for the constants here, as this if statement is very hard to read. 
It would also be better to reverse the condition, as it would be easer to understand, eg.:

```java
if (config.extists() && (config.get(KEY_AUTH_STRATEGY).exists() || ...) {
  return OciConfig.create(config);
}

// here the current content of the if statement
```

Reply:
pushed